internal/xcrd/composite.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package composite
+package xcrd
```

Comment:
i don't think this is a "great" place for this functionality either - this package has been all about generated CRDs to represent XRs and Claims, but i'm not sure I can find a better place either. The reason it has to move, to make sure I understand, is that composite controller package depends on names package, which depends on composition resource name stuff from composite controller package, i.e. circular dependency?

could this stuff just live in the names package instead?

Reply:
yes, to avoid a circular dep. Tt is not related to names though, it is related to the workings of xrd/xr. 

If it were up to me, this would be in the api package for compositions.